public/app/plugins/panel/heatmap-new/utils.ts

```diff
@@ -460,3 +468,64 @@ export const countsToFills = (u: uPlot, seriesIdx: number, palette: string[]) =>
 
   return indexedFills;
 };
+
+export const getHeatmapFields = (dataFrame: DataFrame): Array<Field | undefined> => {
+  const xField: Field | undefined = dataFrame.fields.find((f) => f.name === 'xMin');
+  const yField: Field | undefined = dataFrame.fields.find((f) => f.name === 'yMin');
+  const countField: Field | undefined = dataFrame.fields.find((f) => f.name === 'count');
+
+  return [xField, yField, countField];
+};
+
+export const getHeatmapArrays = (dataFrame: DataFrame): Array<number[] | undefined> => {
+  const [xField, yField, countField] = getHeatmapFields(dataFrame);
+  return [xField?.values.toArray(), yField?.values.toArray(), countField?.values.toArray()];
+};
+
+export const getDataMapping = (
+  heatmapData: HeatmapData,
+  origData: DataFrame,
```

Comment:
`origData` implies that it is the original data that created the heatmap.  This should not be true.  Can we call this "sampleData"?  or maybe "rawValues"?

Reply:
True. I'll make that change.